controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -614,49 +614,48 @@ private Scope getScope(final String scopeName) {
-                                        new AbstractMap.SimpleEntry<>(
-                                                segment.getKeyStart(),
-                                                segment.getKeyEnd())))
-                        .collect(Collectors.toList()));
-    }
-
-    private CompletableFuture<Long> findNumSplits(String scopeName, String streamName, Executor executor) {
+    private CompletableFuture<Pair<Long, Long>> findNumSplitsMerges(String scopeName, String streamName, Executor executor) {
         return getScaleMetadata(scopeName, streamName, null, executor).thenApply(scaleMetadataList -> {
-            long size = scaleMetadataList.size();
+            int size = scaleMetadataList.size();
             long totalNumSplits = 0;
-
-            for (int i = 0, j = 1; j < size; j++) {
-                long countSplitsEventI = scaleMetadataList.get(i).getSegments().size();
-                long countSplitsEventJ = scaleMetadataList.get(j).getSegments().size();
-                totalNumSplits = (countSplitsEventI < countSplitsEventJ) ? (countSplitsEventJ - countSplitsEventI) : 0;
+            long totalNumMerges = 0;
+            List<Segment> segmentList1;
+            List<Segment> segmentList2;
+            boolean isDescendingOrder = (size > 1) ?
+                    (scaleMetadataList.get(0).getTimestamp() < scaleMetadataList.get(1).getTimestamp()) : true;
+
+            for (int i = 0; i < size - 1; i++) {
+                segmentList1 = scaleMetadataList.get(i).getSegments();
+                segmentList2 = scaleMetadataList.get(i+1).getSegments();
+                if (isDescendingOrder) {
+                    totalNumSplits += findSegmentSplitsMerges(segmentList2, segmentList1);
+                    totalNumMerges += findSegmentSplitsMerges(segmentList1, segmentList2);
+                } else {
+                    totalNumSplits += findSegmentSplitsMerges(segmentList1, segmentList2);
+                    totalNumMerges += findSegmentSplitsMerges(segmentList2, segmentList1);
+                }
             }
 
-            return totalNumSplits;
+            return new ImmutablePair<>(totalNumSplits, totalNumMerges);
         });
     }
 
-    private CompletableFuture<Long> findNumMerges(String scopeName, String streamName, Executor executor) {
-        return getScaleMetadata(scopeName, streamName, null, executor).thenApply(scaleMetadataList -> {
-            long size = scaleMetadataList.size();
-            long totalNumMerges = 0;
-
-            for (int i = 0, j = 1; j < size; j++) {
-                long countMergesEventI = scaleMetadataList.get(i).getSegments().size();
-                long countMergesEventJ = scaleMetadataList.get(j).getSegments().size();
-                totalNumMerges = (countMergesEventI > countMergesEventJ) ? (countMergesEventI - countMergesEventJ) : 0;
+    private int findSegmentSplitsMerges(List<Segment> segmentsList1, List<Segment> segmentsList2) {
+        int count = 0;
```

Comment:
However, there is a more important question is of semantics.. 
how do we define splits and merges when its a remap of m segments into n.. 
[note: in disjoint splits and merges its obvious].. 

but we should agree on a definition in case where m segments remap into n segments. 

We will take a simple example and run the above algo and check the output: 
Take for example 2 segments 
i: S0 [0, 0.5), S1 [0.5, 1.0) 
i +1: S2[0.0, 0.33), S3[0.33, 0.66), S4[0.66, 1.0)

(forward computation - `i `and `i+1`)
S0 has 2 overlaps. S1 has 2 overlaps. 
splits = 2 (S0'overlaps -1 + S1'overlaps -1). 

{reverse computation- `i+1` and `i`}
S2 has 1 overlap, S3 has 2 overlaps, S4 has 1 overlap
merges = (S2'overlap - 1 + S3'overlaps - 1 + S4'overlap - 1) = 1

Now is this definition and result along the expected lines. Depends on the definition of splits and merges.